lnd.go

```diff
@@ -78,6 +82,30 @@ func lndMain() error {
 	}
 	defer chanDB.Close()
 
+	// Only process macaroons if --no-macaroons isn't set.
+	var macaroonService *bakery.Service
+	if !cfg.NoMacaroons {
+		// Create the macaroon authentication/authorization service.
+		macaroonService, err = macaroons.NewService(cfg.DataDir)
+		if err != nil {
+			srvrLog.Errorf("unable to create macaroon service: %v",
+				err)
+			return err
+		}
+
+		// Create macaroon files for lncli to use if they don't exist.
+		if !fileExists(cfg.AdminMacPath) &&
```

Comment:
The original logic behind this was that you could delete the admin macaroon and leave a read-only one without an admin macaroon being regenerated. If you still prefer I change this to a `||`, I will do so.